### PR TITLE
WalletProtobufSerializer: migrate to `Network` from `NetworkParameters`

### PR DESCRIPTION
**Pull request by schildbach**
Followup to #3049

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java [hunk 1]
```diff
@@ -446,10 +447,10 @@ public Wallet readWallet(InputStream input, boolean forceReset, @Nullable Wallet
         try {
             Protos.Wallet walletProto = parseToProto(input);
             final String paramsID = walletProto.getNetworkIdentifier();
-            NetworkParameters params = BitcoinNetworkParams.fromID(paramsID);
-            if (params == null)
+            Optional<BitcoinNetwork> network = BitcoinNetwork.fromIdString(paramsID);
+            if (!network.isPresent())
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java:451]: This could be simplified with `.orElseThrow()`

**Review by msgilligan — APPROVED** [review]
LGTM. One minor suggestion.

File: core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java:451]: Oh yes. I'm always scared of picking a Java 9+ method when using these `else` methods.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**